staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -105,8 +107,16 @@ func (w *watcher) Watch(ctx context.Context, key string, rev int64, opts storage
 	if opts.ProgressNotify && w.newFunc == nil {
 		return nil, apierrors.NewInternalError(errors.New("progressNotify for watch is unsupported by the etcd storage because no newFunc was provided"))
 	}
+	initialEventsEndBookmarkRequired, err := w.isInitialEventsEndBookmarkRequired(opts)
+	if err != nil {
+		return nil, err
+	}
+	rev, err = w.getStartWatchResourceVersion(ctx, rev, opts)
+	if err != nil {
+		return nil, err
```

Comment:
Let's ensure that it returns api error.

Reply:
it is, wrapped into an internal error by `transformErrorToEvent`